packages/groq-builder/README.md

```diff
@@ -0,0 +1,92 @@
+# `groq-builder`
+
+A **schema-aware**, strongly-typed GROQ query builder.  
+It enables you to use **auto-completion** and **type-checking** for your GROQ queries. 
+
+## Features
+
+- **Schema-aware** - Use your `sanity.config.ts` for auto-completion and type-checking
```

Comment:
I'll expand this section and add some documentation around this.  

Long story short, if needed, you can **manually** write the types from your schema.  
I know this sounds tedious, but in my experience, all "external schemas" should be documented, and if I'm implementing a query, I'd be more than happy to document the external schema as I'm developing.  This would make my code more resilient, would allow me to mock externals easily, and would not be too difficult.

The schema type is pretty simple; the only complexity is with `references`.  Here's a minimal example:
```ts
declare const references: unique symbol;
type Product = {
  _type: "product",
  name: string,
  price: number,
  category: { _type: "reference", [references]: "category" },
};
type Category = {
  _type: "category",
  name: string,
}
export type SchemaConfig = {
  documentTypes: Product | Category;
  referenceSymbol: typeof referenced;
};
```